openpype/hosts/maya/plugins/inventory/connect_yeti_rig.py

```diff
@@ -0,0 +1,132 @@
+            family = get_representation_context(
+                container["representation"]
+            )["subset"]["data"]["family"]
+            try:
+                containers_by_family[family].append(container)
+            except KeyError:
+                containers_by_family[family] = [container]
+
+        # Validate to only 1 source container.
+        source_containers = containers_by_family.get("animation", [])
+        source_containers += containers_by_family.get("pointcache", [])
+        source_container_namespaces = [
+            x["namespace"] for x in source_containers
+        ]
+        message = (
+            "{} animation containers selected:\n\n{}\n\nOnly select 1 of type "
+            "\"animation\" or \"pointcache\".".format(
+                len(source_containers), source_container_namespaces
+            )
+        )
+        if len(source_containers) != 1:
+            self.display_warning(message)
+            return
+
+        source_container = source_containers[0]
+        source_ids = self.index_by_id(source_container)
+
+        # Target containers.
+        target_ids = {}
+        inputs = []
+        for family, containers in containers_by_family.items():
+            if family in ["animation", "pointcache"]:
+                continue
+
+            for container in containers:
+                target_ids.update(self.index_by_id(container))
+
+                maya_file = get_representation_path(
+                    get_representation_context(
+                        container["representation"]
+                    )["representation"]
+                )
+                _, ext = os.path.splitext(maya_file)
+                settings_file = maya_file.replace(ext, ".rigsettings")
+                if not os.path.exists(settings_file):
+                    continue
+
+                with open(settings_file) as f:
+                    inputs.extend(json.load(f)["inputs"])
+
+        # Compare loaded connections to scene.
+        for input in inputs:
+            source_node = source_ids.get(input["sourceID"])
+            target_node = target_ids.get(input["destinationID"])
+
+            if not source_node or not target_node:
+                self.log.debug(
+                    "Could not find nodes for input:\n" +
+                    json.dumps(input, indent=4, sort_keys=True)
+                )
+                continue
+
+            cmds.connectAttr(
+                "{}.{}".format(source_node, input["connections"][0]),
+                "{}.{}".format(target_node, input["connections"][1])
+            )
+
+    def index_by_id(self, container):
+        reference_node = cmds.sets(container["objectName"], query=True)
+        members = cmds.referenceQuery(reference_node, nodes=True, dagPath=True)
+
+        ids = {}
+        for member in members:
+            if not cmds.objExists(member + ".cbId"):
+                continue
+            ids[cmds.getAttr(member + ".cbId")] = member
```

Comment:
Use the [`openpype.hosts.maya.api.lib.get_id`](https://github.com/ynput/OpenPype/blob/develop/openpype/hosts/maya/api/lib.py#L1272-L1297) function.
```suggestion
            id = lib.get_id(member)
            if not id:
                continue
            ids[id] = member
```

Or:
```suggestion
            ids = {lib.get_id(member): member for member in members}
            ids.pop(None, None)  # ignore anything that didn't have an id
```

Reply:
Cool done